x-pack/legacy/plugins/snapshot_restore/public/app/sections/restore_snapshot/restore_snapshot.tsx

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useEffect, useState } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import { EuiPageBody, EuiPageContent, EuiSpacer, EuiTitle } from '@elastic/eui';
+
+import { SnapshotDetails, RestoreSettings } from '../../../../common/types';
+import { BASE_PATH } from '../../constants';
+import { SectionError, SectionLoading, RestoreSnapshotForm } from '../../components';
+import { useAppDependencies } from '../../index';
+import { breadcrumbService } from '../../services/navigation';
+import { useLoadSnapshot, executeRestore } from '../../services/http';
+
+interface MatchParams {
+  repositoryName: string;
+  snapshotId: string;
+}
+
+export const RestoreSnapshot: React.FunctionComponent<RouteComponentProps<MatchParams>> = ({
+  match: {
+    params: { repositoryName, snapshotId },
+  },
+  history,
+}) => {
+  const {
+    core: { i18n },
+  } = useAppDependencies();
+  const { FormattedMessage } = i18n;
+
+  // Set breadcrumb
+  useEffect(() => {
+    breadcrumbService.setBreadcrumbs('restoreSnapshot');
+  }, []);
+
+  // Snapshot details state with default empty snapshot
+  const [snapshotDetails, setSnapshotDetails] = useState<SnapshotDetails | {}>({});
+
+  // Load snapshot
+  const { error: snapshotError, loading: loadingSnapshot, data: snapshotData } = useLoadSnapshot(
+    repositoryName,
+    snapshotId
+  );
+
+  // Update repository state when data is loaded
+  useEffect(
+    () => {
+      if (snapshotData) {
+        setSnapshotDetails(snapshotData);
+      }
+    },
+    [snapshotData]
+  );
+
+  // Saving repository states
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+  const [saveError, setSaveError] = useState<any>(null);
+
+  // Execute restore
+  const onSave = async (restoreSettings: RestoreSettings) => {
+    setIsSaving(true);
+    setSaveError(null);
+    const { error } = await executeRestore(repositoryName, snapshotId, restoreSettings);
+    if (error) {
+      setIsSaving(false);
+      setSaveError(error);
+    } else {
+      // Wait a few seconds before redirecting so that restore information has time to
+      // populate into master node
+      setTimeout(() => {
+        setIsSaving(false);
+        history.push(`${BASE_PATH}/recovery`);
```

Comment:
Would it make sense (or even be possible) to pass a filter via a query param that can be used to filter the table down to the specific indices being recovered? If a user has a ton of data and is recovering from a catastrophic event then I can imagine them restoring multiple snapshots at once.

Reply:
there are several ES restrictions that prevent us from doing this:
1) selected indices will not all be in progress at once. for example, when user initially hits this page, maybe only 3 indices will show as In Progress. then when ES finishes those, the next poll will retrieve the next batch of indices to restore
2) if user recovers from multiple snapshots at once, then it is possible that one index is being restored from multiple snapshot sources. ES APIs does not return information on a "per-recovery request" basis, so we only have overall index recovery information, not information specifically about the restore they just requested to perform